File: funcoes/Funcoes.java
package funcoes;

class Funcoes {
    public static void main(String[] args) {
        printar("Olá");

        printarComPrefixo("Olá", "Log");
        
        System.out.println(soma(1, 2));
        System.out.println(multiplica(2, 4));
    }

    public static void printar(String mensagem) {
        System.out.println(mensagem);
    }

    public static void printarComPrefixo(String mensagem, String prefixo) {
        System.out.println("[" + prefixo + "] " + mensagem);
    }

    public static int soma(int x, int y) {
        return x + y;
    }

    public static int multiplica(int x, int y) {
        return x * y;
    }
}

File: funcoes/Funcoes.kt
// Unit == void
fun printar(mensagem: String): Unit {
    println(mensagem)
}

// Não existem parâmetros opcionais em java, teria que fazer um overload
// Unit implícito
fun printarComPrefixo(mensagem: String, prefixo: String = "Info") {
    println("[$prefixo] $mensagem")
}

// Função 'clássica'
fun soma(x: Int, y: Int): Int {
    return x + y
}

// Função com retorno 'Int' implícito
fun multiplica(x: Int, y: Int) = x * y

fun main() {
    printar("Olá")

    printarComPrefixo("Olá", "Log")
    printarComPrefixo("Olá")

    printarComPrefixo(prefixo = "Log", mensagem = "Olá")
    
    println(soma(1, 2))
    println(multiplica(2, 4))
}